### PR TITLE
Implemented resolution spin-boxes in the confocal GUI

### PR DESCRIPTION
**Pull request by marcocapelli**
## Description
Added a new set of spin-boxes to set the confocal resolution directly in µm (or any kind of length scale used in the experiment). The changes are mostly in GUI appearance. The code automatically converts everything back in pixels so that the logic maintains the same functionality.

## Motivation and Context
It is often convenient, especially when zooming in and out, to maintain the same spatial resolution and let the computer calculates the correct number of pixels to achieve it.

## How Has This Been Tested?
Tested with the dummy module on a Linux machine.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] I have documented my changes in the changelog (`documentation/changelog.md`)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added/updated for the module the config example in the docstring of the class accordingly.
- [x] I have checked that the change does not contain obvious errors (syntax, indentation, mutable default values).
- [x] I have tested my changes using 'Load all modules' on the default dummy configuration with my changes included.
- [ ] All changed Jupyter notebooks have been stripped of their output cells.

### CODE REVIEW

**Comment by marcocapelli** [comment]
Ok, now you should be able to checkout the changes from the 'confocal-small-adjustment' branch from the CNBP-RMIT repository.

**Review by lishuo8779 — COMMENTED** [review]
Fantastic! Works fine for me. Could you please enlarge the text boxes a little to show the full unit (um/nm instead of mu/n)? Also the "Scan Range" can only show half /mu on my Mac. Probably decrease the size of sliding bars in Scan Range to make some room?

File: gui/confocal/ui_confocalgui.ui [hunk 1]
```diff
@@ -1212,6 +1155,268 @@
...
+       </property>
+       <property name="checked">
+        <bool>false</bool>
+       </property>
+      </widget>
+     </item>
+     <item row="5" column="2">
+      <widget class="QRadioButton" name="length_res_RadioButton">
+       <property name="text">
+        <string>length</string>
```
Comment (MdevriesPhys) [inline gui/confocal/ui_confocalgui.ui]: Suggest "size" instead of "length"

**Comment by MdevriesPhys** [comment]
image

> Also the "Scan Range" can only show half /mu on my Mac. Probably decrease the size of sliding bars in Scan Range to make some room?

I'm seeing a similar issue, also can't resize this widget to view it all.

**Comment by MdevriesPhys** [comment]
> Could you please enlarge the text boxes a little to show the full unit (um/nm instead of mu/n)?

Don't think this is an issue with the size, but more about how the units are implemented in the code - is this a big need? We understand what they mean, right?

**Comment by lishuo8779** [comment]
> > Could you please enlarge the text boxes a little to show the full unit (um/nm instead of mu/n)?
> > Don't think this is an issue with the size, but more about how the units are implemented in the code - is this a big need? We understand what they mean, right?

Sure. It's just a minor issue triggers my obsessiveness...

**Comment by marcocapelli** [comment]
Ok, I made a small mess of things. But now some of the problems should be fixed. Check if the boxes are of the correct size now.

**Comment by MdevriesPhys** [comment]
Still can't seem to resize, seems like no change for me

**Comment by lishuo8779** [comment]
> Still can't seem to resize, seems like no change for me

Me either.
Screen Shot 2020-02-27 at 2 36 54 pm

**Comment by marcocapelli** [comment]
Aaaand yet another try. Looking into the xml code itself there is no indication about the minimum vertical size and all other parameters are the same as the original master branch from Ulm.
Additionally to me, the widget is still not resizable when the GUI is opened the first time, then if I move the widget around (i.e. above the xy map and back down) it becomes resizable again but only by a small amount. The maximum height of the widget is 170points by default in Ulm master branch.

**Comment by MdevriesPhys** [comment]
Made some slight changes to the max/min sizes permitted, should have sorted that issue of not displaying properly/cutting off parts of the text.

**Review by MdevriesPhys — APPROVED** [review]

**Comment by marcocapelli** [comment]
The changes mess up the interface in my computer now:

QuDI_sizes

**Comment by marcocapelli** [comment]
One more change. I kept the minimum sizes (I even increased from 10 to 20) and applied the same size policy to all the science spin boxes (the ones containing numbers and units).
If this still creates problems, I will try to solve it directly on the lab computer since that is going to be the ground truth anyway.

Also, I wanted to test this on the lab computer before approving it. Just to check the real scanning piezo is not pulling weird jokes on me.

**Review by lishuo8779 — APPROVED** [review]

**Pull request merged**